tests/test_download.py

```diff
@@ -165,14 +165,30 @@ def test_mismatching_md5sums(self, tmpfile):
     #
     # Tests for 'singularity_pull_image'
     #
-    # If Singularity is not installed, will log an error and exit
-    # If Singularity is installed, should raise an OSError due to non-existant image
+    # If Singularity is installed, but the container can't be accessed because it does not exist or there are aceess
+    # restrictions, a FileNotFoundError is raised due to the unavailability of the image.
+    @pytest.mark.skipif(
+        shutil.which("singularity") is None,
+        reason="Can't test what Singularity does if it's not installed.",
+    )
     @with_temporary_folder
-    @pytest.mark.xfail(raises=OSError)
     @mock.patch("rich.progress.Progress.add_task")
-    def test_singularity_pull_image(self, tmp_dir, mock_rich_progress):
+    def test_singularity_pull_image_singularity_installed(self, tmp_dir, mock_rich_progress):
         download_obj = DownloadWorkflow(pipeline="dummy", outdir=tmp_dir)
-        download_obj.singularity_pull_image("a-container", tmp_dir, None, mock_rich_progress)
+        with pytest.raises(FileNotFoundError):
+            download_obj.singularity_pull_image("a-container", tmp_dir, None, mock_rich_progress)
+
+    # If Singularity is not installed, it raises a FileNotFoundError because the singularity command can't be found.
+    @pytest.mark.skipif(
+        shutil.which("singularity") is not None,
+        reason="Can't test how the code behaves if sungularity is not installed if it is.",
```

Comment:
I think here you want to say if it is installed, isn't it?
```suggestion
        reason="Can't test how the code behaves if sungularity is installed.",
```

Reply:
No, the point is to say that the code that runs when sigularity is not installed can not be tested if it is installed. The difference is subtle, but the current text makes it clear we are talking about code that runs when singularity is not installed.

Reply:
Ah, OK! I get it now, a bit of a confusing sentence 😄 
LGTM then :)

Reply:
Yes, not entirely happy about it myself.

Reply:
What about using 'when'?
```suggestion
        reason="Can't test how the code behaves when sungularity is not installed if it is.",
```

Reply:
That fits well!